src/main/java/by/exadel/internship/service/impl/TimeForCallUserServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+import java.util.UUID;
+
+;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TimeForCallUserServiceImpl implements TimeForCallUserServise {
+
+    private final TimeForCallUserRepository timeForCallUserRepository;
+    private final UserService userService;
+    private final TimeForCallUserMapper mapper;
+
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_ZERO_THIRTY = 30;
+    private static final int DEFAULT_START_MINUTES_IF_BETWEEN_THIRTY_ZERO = 0;
+    private static int INTERVIEW_TIME = 30;
+    private static final String SIMPLE_CLASS_NAME = TimeForCallUserServise.class.getSimpleName();
+
+
+    private List<TimeForCallWithUserDTO> resultUSerTimeList;
+
+    @Override
+    public void saveUserTime(UserDTO userDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save free time list for user with uuid = {} in DB", userDTO.getId());
+        resultUSerTimeList = new ArrayList<>();
+        userDTO.getTimeForCall().forEach(timeForCallUser -> {
+            checkTime(timeForCallUser);
+            separateTime(timeForCallUser, userDTO.getInterviewTime());
+        });
+        resultUSerTimeList.forEach(time -> {
+            time.setUser(userDTO);
+        });
+        timeForCallUserRepository.saveAll(mapper.mapToEntity(resultUSerTimeList));
+        log.info("Free time list was saved in DB, user uuid = {}", userDTO.getId());
+    }
+
+    @Override
+    public void deletedById(UUID timeId) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to remove time with uuid = {} from user free time", timeId);
+        timeForCallUserRepository.deleteById(timeId);
+        log.info("Time with uuid = {} was deleted", timeId);
+    }
+
+    @Override
+    public void restoreUserTime(TimeForCallWithUserIdDTO time) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Restoring time with uuid = {} for user with uuid ={}", time.getId(), time.getUserId());
+        UserDTO userDTO = userService.getById(time.getUserId());
+        TimeForCallWithUserDTO timeForCallWithUserDTO = new TimeForCallWithUserDTO();
+        timeForCallWithUserDTO.setUser(userDTO);
+        timeForCallWithUserDTO.setStartHour(time.getStartHour());
+        switch (userDTO.getInterviewTime()) {
+            case HALF_HOUR: {
+                timeForCallWithUserDTO.setEndHour(time.getStartHour().plusMinutes(30));
+                break;
+            }
+            case HOUR: {
+                timeForCallWithUserDTO.setEndHour(time.getStartHour().plusMinutes(60));
+                break;
+            }
+            case HOUR_HALF: {
+                timeForCallWithUserDTO.setEndHour(time.getStartHour().plusMinutes(90));
+                break;
+            }
+        }
+        timeForCallUserRepository.save(mapper.toTimeForCallUserEntity(timeForCallWithUserDTO));
+        log.info("User time was restoring");
+    }
+
+    private void checkTime(TimeForCallUserDTO time) {
```

Comment:
This method can be simplified. 
Provide additional method which will work with LocalDateTime (so you can send StartHour and EndHour). 
Then you can check minutes is divided by 60 with the remainder or not. And use methods  `plusMinutes(long)` and `minusMinutes(long)`

Reply:
Done